In [ ]:
here::i_am("readme.md")
library(here)
library(dplyr)

here() starts at C:/Users/Plancha/Desktop/PACD



O nosso objetivo é de prever o número de *sets* que um jogo de ténis profissional de Austrália vai ter, baseado num número de recursos que um jogo vai ter. 

Um jogo de ténis é composto por pontos, jogos e sets. De acordo com as [regras de ténis da Federação Internacional de Tennis (IFT)](https://www.atptour.com/en/corporate/rulebook), um set é uma pontuação, onde o primeiro jogador a ganhar 6 jogos ganha um set, sendo um jogo decidido pelo número de sets que um jogador ganha. Um jogo pode ser à melhor de 3 ou à melhor de 5, ou seja, o jogador precisa de ganhar 2 sets (o que traduz para 3 sets de máximo) ou 3 sets (máximo 5), respectivamente.  O jogador que decide o lado que começa/quem serve primeiro é decidido aleatoriamente, sendo que o outro decide a outra decisão.

Baseado nestes factos, parece que o projeto vai ser de regressão ordinal, sendo que o número de sets varia entre 2 e 5, não mais e não menos.

In [ ]:
arrow::read_parquet(here("data", "atpRaw.parquet"), as_data_frame = TRUE) -> atp
head(atp)

Born             Date                    GameRank GameRound   Ground
1 Belgrade, Serbia 2021.11.22 - 2021.12.05 30       Semi-Finals Hard  
2 Belgrade, Serbia 2022.02.21 - 2022.02.26 26       Round of 16 Hard  
3 Belgrade, Serbia 2021.11.15 - 2021.11.21 12       Round Robin Hard  
4 Belgrade, Serbia 2021.11.15 - 2021.11.21 5        Round Robin Hard  
5 Belgrade, Serbia 2021.11.01 - 2021.11.07 2        Finals      Hard  
6 Belgrade, Serbia 2021.11.01 - 2021.11.07 10       Semi-Finals Hard  
  Hand                              Height
1 Right-Handed, Two-Handed Backhand 188   
2 Right-Handed, Two-Handed Backhand 188   
3 Right-Handed, Two-Handed Backhand 188   
4 Right-Handed, Two-Handed Backhand 188   
5 Right-Handed, Two-Handed Backhand 188   
6 Right-Handed, Two-Handed Backhand 188   
  LinkPlayer                                                                                       
1 https://www.atptour.com/en/players/novak-djokovic/d643/player-activity?year=all&matchType=Singles
2 https://www.atptour.com/en/players/novak-djokovic/d643/player-activity?year=all&matchType=Singles
3 https://www.atptour.com/en/players/novak-djokovic/d643/player-activity?year=all&matchType=Singles
4 https://www.atptour.com/en/players/novak-djokovic/d643/player-activity?year=all&matchType=Singles
5 https://www.atptour.com/en/players/novak-djokovic/d643/player-activity?year=all&matchType=Singles
6 https://www.atptour.com/en/players/novak-djokovic/d643/player-activity?year=all&matchType=Singles
  Location      Oponent         PlayerName     Prize      Score   
1 Madrid, Spain Marin Cilic     Novak Djokovic            64 62   
2 Dubai, U.A.E. Karen Khachanov Novak Djokovic $2,794,840 63 76   
3 Turin, Italy  Cameron Norrie  Novak Djokovic $7,250,000 62 61   
4 Turin, Italy  Andrey Rublev   Novak Djokovic $7,250,000 63 62   
5 Paris, France Daniil Medvedev Novak Djokovic �5,207,405 46 63 63
6 Paris, France Hubert Hurkacz  Novak Djokovic �5,207,405 36 60 76
  Tournament             WL _id.$oid                
1 Davis Cup Finals       W  624ab34913b144c54b3c9aba
2 Dubai                  W  624ab34913b144c54b3c9abb
3 Nitto ATP Finals       W  624ab34913b144c54b3c9abc
4 Nitto ATP Finals       W  624ab34913b144c54b3c9abd
5 ATP Masters 1000 Paris W  624ab34913b144c54b3c9abe
6 ATP Masters 1000 Paris W  624ab34913b144c54b3c9abf

In [ ]:
nrow(atp)

[1] 1308835

A base de dados mostra informação pública de [ATP Tour](https://www.atptour.com/en/players).

A primeira vista dos dados revela vários factos:

* A base de dados tem 1308835, sendo cada registro um jogo de um jogador "principal" (com as informações de tal) contra um oponente (sem as informações de tal).
    * Isto poderá indicar que teremos de transformar os dados em registros de jogos entre os dois jogadores, sem jogador principal. Não temos a certeza se isto é necessário.
* Ao fazer a cabeça dos dados, parece que apenas mostra o jogador Novak Djokovic como principal. Isto pode indicar que a base de dados pode estar ordenada de uma certa forma.
    * Esta ordem apenas pode indicar como os dados foram obtidos, e nada sobre um registro específico. Informação sobre a recolha pode ajudar a perceber melhor o contexto da base de dados.
* A data parece mostrar as datas do torneio e não do jogo especificamente
* A coluna `prize` parece mal formatada, com valores omissos presentes. Da mesma forma, aprece que os valores estão em várias moedas diferentes.
* A coluna `GameRank` parece indicar o *rank* do oponente, não o rank do jogo.
    * Desta forma, desde que o oponente apareça também como principal, consegue-se ter o rank de ambas as pessoas. No entanto, isto não é garantido.
    * Nos casos que isto acontece, pode ajudar fazer a diferença de rank entre jogadores.
* A coluna `Hand` parece incluir ambas as informações do *backhand* e se é destro ou esquerdino
* O score parece conter a pontuação de cada set, sendo ai onde vamos buscar o número de sets.

In [ ]:
set.seed(1)
atp %>% slice_sample(n=5)

Born              Date                    GameRank GameRound           
1                   2010.08.02 - 2010.08.08 -        Semi-Finals         
2 Sendai, Japan     2017.03.13 - 2017.03.19 164      Round of 16         
3                   2005.11.21 - 2005.11.27 1367     Round of 32         
4 Voorhees, NJ, USA 2015.04.20 - 2015.04.26 462      2nd Round Qualifying
5 Santiago, Chile   2007.03.19 - 2007.04.01 107      1st Round Qualifying
  Ground Hand                              Height
1 Clay   null                              NA    
2 Hard   Right-Handed, Two-Handed Backhand 175   
3 Hard   null                              NA    
4 Clay   Right-Handed, Two-Handed Backhand 185   
5 Hard   Right-Handed, Two-Handed Backhand 191   
  LinkPlayer                                                                                         
1 https://www.atptour.com/en/players/juan-lizariturry/l996/player-activity?year=all&matchType=Singles
2 https://www.atptour.com/en/players/yuichi-sugita/se73/player-activity?year=all&matchType=Singles   
3 https://www.atptour.com/en/players/ludwig-pellerin/pa58/player-activity?year=all&matchType=Singles 
4 https://www.atptour.com/en/players/tommy-paul/pl56/player-activity?year=all&matchType=Singles      
5 https://www.atptour.com/en/players/paul-capdeville/c634/player-activity?year=all&matchType=Singles 
  Location             Oponent          PlayerName       Prize      Score     
1 Spain                Taro Daniel      Juan Lizariturry $10,000    63 62     
2 Shenzhen, China      Jan Satral       Yuichi Sugita    $75,000    67,  63 62
3 Monastin, Tunisia    Anas Fattar      Ludwig Pellerin  $10,000    64 64     
4 Savannah, GA, U.S.A. Andrea Collarini Tommy Paul       $50,000    76,  62   
5 Miami, FL, U.S.A.    Andrei Pavel     Paul Capdeville  $3,200,000 26 63 75  
  Tournament             WL _id.$oid                
1 Spain F28              W  624ab365a5c952847318a12b
2 Shenzhen               W  624ab34f936e4d97ffee0700
3 Tunisia F5             W  624ab40fba7043df73e7f9e9
4 Savannah               W  624ab34a13b144c54b3cfd0e
5 ATP Masters 1000 Miami W  624ab37573539d8f27174222

Estes dados mostram também que há jogadores principais sem informações.

In [ ]:
library(stringr)
atp %>% distinct(Location) %>% filter(str_like(Location, "%Aus%") & !str_like(Location, "%Austria%"))



processing file: report.rmd



  |..........................           |  71%                                  (global_options)                                                  (introduce)                                                       (plot_intro)                                                      (data_structure)                                                  (missing_profile)                                                 (univariate_distribution_header)                                  (plot_histogram)                                                  (plot_density)                                                    (plot_frequency_bar)                                              (plot_response_bar)                                               (plot_with_bar)                                                   (plot_normal_qq)                                                  (plot_response_qq)                                                (plot_by_qq)                                                  

In [ ]:
atp %>% distinct(PlayerName)

PlayerName                   
1    Novak Djokovic               
2    Daniil Medvedev              
3    Rafael Nadal                 
4    Alexander Zverev             
5    Stefanos Tsitsipas           
6    Matteo Berrettini            
7    Andrey Rublev                
8    Casper Ruud                  
9    Felix Auger-Aliassime        
10   Hubert Hurkacz               
11   Jannik Sinner                
12   Cameron Norrie               
13   Taylor Fritz                 
14   Denis Shapovalov             
15   Diego Schwartzman            
16   Roberto Bautista Agut        
17   Reilly Opelka                
18   Pablo Carreno Busta          
19   Nikoloz Basilashvili         
20   Lorenzo Sonego               
21   John Isner                   
22   Marin Cilic                  
23   Gael Monfils                 
24   Karen Khachanov              
25   Roger Federer                
26   Daniel Evans                 
27   Alex de Minaur               
28   Grigor Dimitrov              
29   Cristian Garin               
30   Frances Tiafoe               
⋮    ⋮                            
9924 Patrick Zahraj               
9925 Pedro Boscardin Dias         
9926 Petar Pjevcevic              
9927 Philip Bachmaier             
9928 Philip Henning               
9929 Preston Brown                
9930 Quentin Gueydan              
9931 Qun Gao                      
9932 Rayane Oumaouche             
9933 Riccardo Di Nocera           
9934 Robert Galloway              
9935 Roi Ginat                    
9936 Roko Horvat                  
9937 Ryotero Matsumura            
9938 Saba Purtseladze             
9939 Sahar Simon                  
9940 Sam Taylor                   
9941 Sanetulla Kutibaev           
9942 Sarp Agabigun                
9943 Savva Polukhin               
9944 Sean Hodkin                  
9945 Sergio Galdos                
9946 Sergio Luis Hernandez Ramirez
9947 Shahin Khaledan              
9948 Sheng Hao Zhou               
9949 Shunsuke Wakita              
9950 Sven Lah                     
9951 Tadas Babelis                
9952 Toufik Sahtali               
9953 Volodoymyr Uzhylovskyi